service/share/cache_availability.go

```diff
@@ -50,18 +53,22 @@ func (ca *CacheAvailability) SharesAvailable(ctx context.Context, root *Root) er
 	}
 	// do not sample over Root that has already been sampled
 	key := rootKey(root)
+
+	ca.dsLk.Lock()
```

Comment:
RLock?

Reply:
Oh yeah lol. I thought I made an Rlock there. I remember looking at source of has to check if it doesn't do any writes in the background. Thanks for catching